packages/x-charts-pro/src/FunnelChart/funnel.types.ts

```diff
@@ -0,0 +1,143 @@
+import { ChartsLabelMarkProps } from '@mui/x-charts/ChartsLabel';
+import {
+  CommonSeriesType,
+  CartesianSeriesType,
+  CommonDefaultizedProps,
+  SeriesId,
+} from '@mui/x-charts/internals';
+import { CurveType, Position } from '@mui/x-charts/models';
+import { DefaultizedProps } from '@mui/x-internals/types';
+
+export type FunnelItemId = string | number;
+
+export type FunnelCurveType = Extract<CurveType, 'linear' | 'step' | 'bumpY' | 'bumpX'>;
+
+export type FunnelValueType = {
+  /**
+   * A unique identifier of the funnel section.
+   */
+  id?: FunnelItemId;
+  /**
+   * The value of the funnel section.
+   */
+  value: number;
+  /**
+   * The label to display on the tooltip, funnel section, or the legend. It can be a string or a function.
+   */
+  label?: string | ((location: 'tooltip' | 'legend' | 'section') => string);
+  /**
+   * The color of the funnel section
+   */
+  color?: string;
+  /**
+   * Defines the mark type for the funnel item.
+   * @default 'square'
+   */
+  labelMarkType?: ChartsLabelMarkProps['type'];
+};
+
+export interface FunnelSeriesType
+  extends Omit<CommonSeriesType<FunnelValueType>, 'color'>,
+    CartesianSeriesType {
+  type: 'funnel';
+  /**
+   * Data associated to the funnel section.
+   */
+  data?: Readonly<FunnelValueType[]>;
+  /**
+   * The label to display on the tooltip or the legend. It can be a string or a function.
+   */
+  label?: string | ((location: 'tooltip' | 'legend') => string);
+  /**
+   * Layout of the funnel.
+   * @default 'vertical'
+   */
+  layout?: 'horizontal' | 'vertical';
```

Comment:
Does it make sense for the `layout` to be a prop of the series instead of the chart? 

We can have multiple series (by using stacking). What should happen if the first series is `layout: 'horizontal'`, but the second one is `layout: 'vertical'`?

Reply:
It is currently a prop of the series. Do you mean to say it should be a prop of the `chart`? Possibly, I'll still refine this API, that could be a nice change. 

Currently the API is focused to be similar to the other charts, but it is not necessary to be so, because the is not much need to compose a funnel and a bar chart for example, so I'll be updating the API to be more focused on funnel needs eventually

Reply:
> Do you mean to say it should be a prop of the chart?

Yes, sorry 😛 I wrote it the other way around.

I guess we can leave it a as a prop of the series, but we would need to handle it properly in that case. From my tests, it seems we aren't properly handling series with different layouts within the same funnel chart.